src/openzeppelin/account/library.cairo

```diff
@@ -175,13 +180,16 @@ namespace Account {
         alloc_locals;
 
         let (tx_info) = get_tx_info();
-        with_attr error_message("Account: invalid tx version") {
-            assert tx_info.version = 1;
+        // Allow query calls for simulations and assert transaction version
+        if (tx_info.version != QUERY_VERSION) {
+            with_attr error_message("Account: invalid tx version") {
+                assert tx_info.version = TRANSACTION_VERSION;
```

Comment:
Should we test that `QUERY_VERSION` txs will fail upon execution? I'm not sure that it's on us to do so, but I'm curious on what you think

Reply:
That's protocol level. I don't think we need to test it.

Reply:
why would `QUERY_VERSION` fail on execution? this might be unrelated but while analyzing this change I realized these accounts are not future proof. the only reason we have the version check is to disallow txs of version 0.

if there were version 2 txs in the future, these accounts would stop working and require a migration, are we sure we want to impose this considering that accounts could be perfectly compatible with such txs?

if we choose to simply disallow version 0, I suggest we replace this double check (if + assert) with `assert_le` to make sure we allow any version > 0.

Reply:
> why would QUERY_VERSION fail on execution?

`QUERY_VERSION` txs should fail if they're not queries. 

> are we sure we want to impose this considering that accounts could be perfectly compatible with such txs?

Great question! It's definitely not future-proof and would require migration in future versions. IMO erring on the side of caution seems wise especially considering the gravitas of the account contract. We don't know (at least I don't) what future tx versions will look like. If tx version 2, for example, includes some new feature that requires another check, this could allow the possibility of using the version 1 model to bypass the version 2 check.

Maybe I'm being overly cautious. What do you guys think?